ui/src/app/applications/components/application-parameters/application-parameters.tsx

```diff
@@ -142,6 +142,268 @@ function getParamsEditableItems(
         .map((item, i) => ({...item, before: (i === 0 && <p style={{marginTop: '1em'}}>{title}</p>) || null}));
 }
 
+function mergeAndFormatHelmValues(valuesObject: any, values: string) {
+    if (!valuesObject && !values) {
+        return '';
+    }
+    //Order of precedence: valuesObject > values
+    const mergedValueObject = merge(jsYaml.load(values || ''), valuesObject || {});
+    return jsYaml.dump(mergedValueObject);
+}
+
+function flattenObject(obj: any, parentPath = '', result = {} as any) {
+    forEach(obj, (value, key) => {
+        const fullPath = parentPath ? (isArray(obj) ? `${parentPath}[${key}]` : `${parentPath}.${key}`) : key;
+        if (isObject(value) || isArray(value)) {
+            flattenObject(value, fullPath, result);
+        } else {
+            result[fullPath] = value;
+        }
+    });
+    return result;
+}
+
+function unflattenedObject(obj: any) {
+    const result = {};
+    for (const key in obj) {
+        if (Object.prototype.hasOwnProperty.call(obj, key)) {
+            set(result, key, obj[key]);
+        }
+    }
+    return result;
+}
```

Comment:
Could we have some unit tests for these utility functions? I think we're trying to mimic Helm's values file merging behavior. If we find later that our merge algorithms differ from Helm's, it would be good to have unit tests to help safely change our algorithm to match Helm's.

Reply:
I completely agree！

Give me some time to review the original Helm's values file merging behavior and add test cases for these functions.

Reply:
I have updated the description of this PR.
This PR now only fix the issue of preventing a crash when switching from YAML to form mode with valuesObject during application creation.

It does not include any changes to value or parameters, as I believe the UI should only be responsible for displaying configuration, not handling the logic of data merging.